src/backend/validate.ts

```diff
@@ -0,0 +1,60 @@
+import { NegatableOption, RequestSpec } from '../RequestSpec';
+import { Edition } from '../types';
+import { labelForRequestSpecKey } from '../ui/utils';
+
+// Type predicate
+// eslint-disable-next-line
+const isNegatableOption = (value: any): value is NegatableOption => {
+  return typeof value.negated === 'boolean';
+};
+
+const isDefaultValue = (value: string | NegatableOption | string[] | undefined): boolean => {
+  if (typeof value === 'undefined') {
+    return true;
+  }
+  if (typeof value === 'string') {
+    return value === '';
+  }
+  if (isNegatableOption(value)) {
+    return value.value !== undefined || value.value !== '';
+  }
+  if (Array.isArray(value)) {
+    return value.length > 0;
```

Comment:
this would mean any array with at least one element is a default value? is this true?
maybe a unittest could help here to clear things up?

Reply:
Thought backward on this one. It's a default value if the array is empty. But anyways, that case is redundant, non of the fields that can be required are arrays.